### PR TITLE
Cleanup of config.py

### PR DESCRIPTION
**Pull request by Athanasius**
The intent is to pass flake8 and mypy tests.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -82,194 +102,283 @@
...
+
             else:
                 return None
 
-        def getint(self, key):
+        def getint(self, key: str) -> int:
+            """Look up an integer configuration value."""
             try:
-                return int(self.settings.get(key, 0))	# should already be int, but check by casting
-            except:
```
Comment (A-UNDERSCORE-D) [inline config.py:163]: `ValueError: invalid literal for int() with base 10: 'asd'` et al is the most likely, I believe.

File: config.py [hunk 2]
```diff
@@ -284,72 +393,86 @@ def __init__(self):
...
                 self.set('outdir', expanduser('~'))
 
-        def get(self, key):
+        def get(self, key: str) -> Union[None, list, str]:
+            """Look up a string configuration value."""
             try:
                 val = self.config.get(self.SECTION, key)
-                if u'\n' in val:	# list
-                    # ConfigParser drops the last entry if blank, so we add a spurious ';' entry in set() and remove it here
+                if u'\n' in val:  # list
```
Comment (A-UNDERSCORE-D) [inline config.py:421]:
```suggestion
                if '\n' in val:  # list
```

File: config.py [hunk 3]
```diff
@@ -284,72 +393,86 @@ def __init__(self):
...
+                    # ConfigParser drops the last entry if blank,
+                    # so we add a spurious ';' entry in set() and remove it here
                     assert val.split('\n')[-1] == ';', val.split('\n')
                     return [self._unescape(x) for x in val.split(u'\n')[:-1]]
```
Comment (A-UNDERSCORE-D) [inline config.py:425]:
```suggestion
                    return [self._unescape(x) for x in val.split('\n')[:-1]]
```

File: config.py [hunk 4]
```diff
@@ -284,72 +393,86 @@ def __init__(self):
...
+            """Set value on the specified configuration key."""
             if isinstance(val, bool):
                 self.config.set(self.SECTION, key, val and '1' or '0')
+
             elif isinstance(val, str) or isinstance(val, numbers.Integral):
                 self.config.set(self.SECTION, key, self._escape(val))
-            elif hasattr(val, '__iter__'):	# iterable
+
+            elif isinstance(val, list):
                 self.config.set(self.SECTION, key, u'\n'.join([self._escape(x) for x in val] + [u';']))
```
Comment (A-UNDERSCORE-D) [inline config.py:451]:
```suggestion
                self.config.set(self.SECTION, key, '\n'.join([self._escape(x) for x in val] + [u';']))
```

Comment (Athanasius) [reply config.py:451]: And presumably `u';'` -> `';'` as well.

Comment (A-UNDERSCORE-D) [reply config.py:451]: Yeah, missed one :D

File: config.py (hunk 1 above)
Comment (Athanasius) [reply config.py:163]: I'll add an except for ValueError, but keep the general one for now, just in case.

File: config.py (hunk 4 above)
Comment (Athanasius) [reply config.py:451]: Doing a search on u' and changing them all in this file....

**Comment by Athanasius** [comment]
I ran into trouble with my pre-commit not liking the f'{var=}' syntax, turns out it needed updating to use Python 3.8, and whilst I was at it I stopped pinning an old version of it, let it use latest.

File: config.py [hunk 5]
```diff
@@ -82,194 +102,288 @@
...
         return retval
 
-
-elif platform=='linux':
+elif platform == 'linux':
     import codecs
     from configparser import RawConfigParser
 
 
 class Config(object):
```
Comment (A-UNDERSCORE-D) [inline config.py:119]:
```suggestion
class Config:
```

No need to subclass object as we dont support py2 anymore

**Review by A-UNDERSCORE-D — APPROVED** [review]

**Pull request merged**